### ISSUE TITLE
Not working after update 1.18f

### ISSUE THREAD

**Issue by Mimonsi** [SonnyX/CS2-TransportPolicyAdjuster#5]
The number of vehicles is set to NaN and the mod cannot be used at all. Can provide logs if needed.

**Comment by Ornithorink** [comment]
I confirm, getting the same error. When trying to see the details of a line, the following errors appear:

```
[object Object]
Colossal.Logging.CustomLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:LogFormat(LogType, String, Object[])
Colossal.Logging.UnityLogger:Log(Level, String, Exception)
Colossal.Logging.UnityLogger:Critical(Object)
TransportPolicyAdjuster.ReflectionExtensions:GetMemberValue(Object, String) (at C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\ReflectionExtensions.cs:34)
TransportPolicyAdjuster.VehicleCountSection:OnUpdate(VehicleCountSection&) (at C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\VehicleCountSection.cs:259)
Game.UI.InGame.VehicleCountSection:Game.UI.InGame.VehicleCountSection.OnUpdate_Patch1(VehicleCountSection)
Unity.Entities.SystemBase:Update()
Game.UI.InGame.InfoSectionBase:PerformUpdate()
Game.UI.InGame.SelectedInfoUISystem:UpdateSections()
Game.UI.InGame.SelectedInfoUISystem:OnUpdate()
Unity.Entities.SystemBase:Update()
Game.UpdateSystem:Update(SystemUpdatePhase)
Game.UI.UIUpdateSystem:OnUpdate()
Unity.Entities.SystemBase:Update()
Game.UpdateSystem:Update(SystemUpdatePhase)
Game.SceneFlow.GameManager:UpdateWorld()
Game.SceneFlow.GameManager:Update()
```
```
[object Object]
Exception: ReflectionExtensions: Couldn't find member name: m_CountResult!
  at TransportPolicyAdjuster.ReflectionExtensions.GetMemberValue[T] (System.Object obj, System.String memberName) [0x00071] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\ReflectionExtensions.cs:36 
  at TransportPolicyAdjuster.VehicleCountSection.OnUpdate (Game.UI.InGame.VehicleCountSection& __instance) [0x0011d] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\VehicleCountSection.cs:259 
Colossal.Logging.CustomLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:LogFormat(LogType, String, Object[])
Colossal.Logging.UnityLogger:Log(Level, String, Exception)
Colossal.Logging.UnityLogger:Critical(Exception, Object)
TransportPolicyAdjuster.VehicleCountSection:OnUpdate(VehicleCountSection&) (at C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\VehicleCountSection.cs:283)
Game.UI.InGame.VehicleCountSection:Game.UI.InGame.VehicleCountSection.OnUpdate_Patch1(VehicleCountSection)
Unity.Entities.SystemBase:Update()
Game.UI.InGame.InfoSectionBase:PerformUpdate()
Game.UI.InGame.SelectedInfoUISystem:UpdateSections()
Game.UI.InGame.SelectedInfoUISystem:OnUpdate()
Unity.Entities.SystemBase:Update()
Game.UpdateSystem:Update(SystemUpdatePhase)
Game.UI.UIUpdateSystem:OnUpdate()
Unity.Entities.SystemBase:Update()
Game.UpdateSystem:Update(SystemUpdatePhase)
Game.SceneFlow.GameManager:UpdateWorld()
Game.SceneFlow.GameManager:Update()
```

![image](https://github.com/user-attachments/assets/11087360-4507-471a-a2f3-0e6b590b4569)


Also happen when trying to change the amount of assigned vehicles.

**Comment by ArcturusZhang** [comment]
So CO must have changed how the `VehicleCountSection` implements.
It looks like they are using these fields to replace the `m_ResultCount`:
```csharp
	private int vehicleCountMin { get; set; }

	private int vehicleCountMax { get; set; }
```

**Comment by BoldArtistix** [comment]
Also receiving the error. Happens on creating new lines, modifying lines, and selecting busses & waiting cims.

```
[2024-09-05 13:24:37,630] [WARN]  m_CountResult's GetMemberInfo returned null
[2024-09-05 13:24:37,647] [CRITICAL]  ReflectionExtensions: Couldn't find member name: m_CountResult!
[2024-09-05 13:24:37,677] [CRITICAL]  Something went wrong in the OnUpdate of VehicleCountSection
Exception: ReflectionExtensions: Couldn't find member name: m_CountResult!
  at TransportPolicyAdjuster.ReflectionExtensions.GetMemberValue[T] (System.Object obj, System.String memberName) [0x00071] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\ReflectionExtensions.cs:36 
  at TransportPolicyAdjuster.VehicleCountSection.OnUpdate (Game.UI.InGame.VehicleCountSection& __instance) [0x0011d] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\VehicleCountSection.cs:259

[2024-09-05 13:24:54,505] [WARN]  m_CountResult's GetMemberInfo returned null
[2024-09-05 13:24:54,522] [CRITICAL]  ReflectionExtensions: Couldn't find member name: m_CountResult!
[2024-09-05 13:24:54,541] [CRITICAL]  Something went wrong in the OnSetVehicleCount of VehicleCountSection
Exception: ReflectionExtensions: Couldn't find member name: m_CountResult!
  at TransportPolicyAdjuster.ReflectionExtensions.GetMemberValue[T] (System.Object obj, System.String memberName) [0x00071] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\ReflectionExtensions.cs:36 
  at TransportPolicyAdjuster.VehicleCountSection.OnSetVehicleCount (System.Single newVehicleCount, Game.UI.InGame.VehicleCountSection& __instance) [0x00027] in C:\Users\randy\git\TransportPolicyAdjuster\TransportPolicyAdjuster\VehicleCountSection.cs:162
```

**Comment by SonnyX** [comment]
Fixed in version 1.4.0 :)
